--- a/bazel/gflags.bzl	Wed Mar 18 13:31:18 2020
+++ b/bazel/gflags.bzl	Fri Sep 26 14:06:38 2025
@@ -49,7 +49,7 @@
         "src/mutex.h",
         "src/util.h",
     ] + select({
-        "//:x64_windows": [
+        "@platforms//os:windows": [
             "src/windows_port.cc",
             "src/windows_port.h",
         ],
@@ -74,7 +74,7 @@
         "-DHAVE_STRTOQ",
         "-DHAVE_RWLOCK",
     ] + select({
-        "//:x64_windows": [
+        "@platforms//os:windows": [
             "-DOS_WINDOWS",
         ],
         "//conditions:default": [
@@ -86,7 +86,7 @@
     linkopts = []
     if threads:
         linkopts += select({
-            "//:x64_windows": [],
+            "@platforms//os:windows": [],
             "//conditions:default": ["-lpthread"],
         })
     else:
