diff --git a/google/cloud/kms/v1/resources.proto b/google/cloud/kms/v1/resources.proto
index 5029f3e03..b32d85464 100644
--- a/google/cloud/kms/v1/resources.proto
+++ b/google/cloud/kms/v1/resources.proto
@@ -403,6 +403,18 @@ message CryptoKeyVersion {
     // HMAC-SHA256 signing with a 256 bit key.
     HMAC_SHA256 = 32;
 
+    // HMAC-SHA1 signing with a 160 bit key.
+    HMAC_SHA1 = 33;
+
+    // HMAC-SHA384 signing with a 384 bit key.
+    HMAC_SHA384 = 34;
+
+    // HMAC-SHA512 signing with a 512 bit key.
+    HMAC_SHA512 = 35;
+
+    // HMAC-SHA224 signing with a 224 bit key.
+    HMAC_SHA224 = 36;
+
     // Algorithm representing symmetric encryption by an external key manager.
     EXTERNAL_SYMMETRIC_ENCRYPTION = 18;
   }
