diff --git a/go/private/repositories.bzl b/go/private/repositories.bzl
index 42b14690..dae616f6 100644
--- a/go/private/repositories.bzl
+++ b/go/private/repositories.bzl
@@ -256,6 +256,8 @@ def go_rules_dependencies(is_rules_go = False):
             "@io_bazel_rules_go//third_party:go_googleapis-directives.patch",
             # gazelle args: -repo_root .
             "@io_bazel_rules_go//third_party:go_googleapis-gazelle.patch",
+            # Patch raw PKCS #1 signing fields into Cloud KMS
+            "@com_google_kmstools//:third_party/googleapis.patch",
         ],
         patch_args = ["-E", "-p1"],
     )
